buddyboss/buddyboss-platform
# dropzonejs custom options

### All Submissions:






### Changes proposed in this Pull Request:


BuddyBoss uses dropzonejs for the activity uploads and it is currently not possible to modify the dropzonejs options, i.e. to enable client side image resizing.

This pull request allows extra options to be added to the BuddyBoss uploader. It uses the already existing javascript global, BP_Nouveau, which can be modified using the core filter bp_core_get_js_strings.

### How to test the changes in this Pull Request:

1. Add filter to set dropzonejs javascript global BP_Nouveau.media.dropzone_options
```
// define the bp_core_get_js_strings callback
function filter_bp_core_get_js_strings( $var = array() ) {
    // add dropzone options
    $var['media']['dropzone_options'] = [
        'resizeWidth' => 1920,
        'resizeHeight' => 1080,
        'resizeQuality' => 0.9,
        'resizeMethod' => 'contain',
        'resizeMimeType' => 'image/jpg'
    ];
    return $var;
};

// add the filter
add_filter( 'bp_core_get_js_strings', 'filter_bp_core_get_js_strings', 100, 1 );
```
2. Upload a large image to an activity post, profile photo, album, group photo, album and forum photo, album
3. Image should be resized on the client side before being uploaded

### Proof Screenshots or Video



### Other information:




### Changelog entry

> Dropzonejs - allow custom options via BP_Nouveau global object
